pkg/kubelet/volume/nfs_mount/nfs_mount.go

```diff
@@ -0,0 +1,169 @@
+limitations under the License.
+*/
+
+package nfs_mount
+
+import (
+	"os"
+	"strings"
+	"syscall"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/kubelet/volume"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/types"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util/mount"
+	"github.com/golang/glog"
+)
+
+// This is the primary entrypoint for volume plugins.
+func ProbeVolumePlugins() []volume.Plugin {
+	return []volume.Plugin{&nfsMountPlugin{nil}}
+}
+
+type nfsMountPlugin struct {
+	host volume.Host
+}
+
+var _ volume.Plugin = &nfsMountPlugin{}
+
+const (
+	nfsMountPluginName = "kubernetes.io/nfs-mount"
+)
+
+func (plugin *nfsMountPlugin) Init(host volume.Host) {
+	plugin.host = host
+}
+
+func (plugin *nfsMountPlugin) Name() string {
+	return nfsMountPluginName
+}
+
+func (plugin *nfsMountPlugin) CanSupport(spec *api.Volume) bool {
+	if spec.Source.NFSMount != nil {
+		return true
+	}
+	return false
+}
+
+func (plugin *nfsMountPlugin) NewBuilder(spec *api.Volume, podUID types.UID) (volume.Builder, error) {
+	return plugin.newBuilderInternal(spec, podUID, mount.New())
+}
+
+func (plugin *nfsMountPlugin) newBuilderInternal(spec *api.Volume, podUID types.UID, mounter mount.Interface) (volume.Builder, error) {
+
+	return &nfsMount{
+		volName:      spec.Name,
+		server:       spec.Source.NFSMount.Server,
+		exportPath:   spec.Source.NFSMount.SourcePath,
+		mountOptions: spec.Source.NFSMount.MountOptions,
+		podUID:       podUID,
+		mounter:      mounter,
+		plugin:       plugin,
+	}, nil
+}
+
+func (plugin *nfsMountPlugin) NewCleaner(volName string, podUID types.UID) (volume.Cleaner, error) {
+	return plugin.newCleanerInternal(volName, podUID, mount.New())
+}
+
+func (plugin *nfsMountPlugin) newCleanerInternal(volName string, podUID types.UID, mounter mount.Interface) (volume.Cleaner, error) {
+	return &nfsMount{
+		volName:      "",
+		server:       "",
+		exportPath:   "",
+		mountOptions: "",
+		podUID:       podUID,
+		mounter:      mounter,
+		plugin:       plugin,
+	}, nil
+}
+
+// NFSMount volumes represent a bare host file or directory mount of an NFS export.
+// The direct at the specified path will be directly exposed to the container.
+type nfsMount struct {
+	volName      string
+	podUID       types.UID
+	server       string
+	exportPath   string
+	mountOptions string
+	mounter      mount.Interface
+	plugin       *nfsMountPlugin
+}
+
+func (nfs *nfsMount) SetUp() error {
+	flags := uintptr(0)
+	if strings.Contains(nfs.mountOptions, "ro") {
+		flags = mount.FlagReadOnly
+	}
+	exportDir := nfs.exportPath
+	mountDir := nfs.GetPath()
+	exists, err := exists(mountDir)
+	if err != nil {
+		glog.Errorf("Mount directory is invalid: %v, %v", mountDir, err)
+	}
+	if !exists {
+		os.MkdirAll(mountDir, 0750)
+	}
+	// NFS Mount format is server:/export/path /mount
+	syscall.Mount(nfs.server+":"+exportDir, mountDir, "nfs", mount.FlagBind|flags, "")
```

Comment:
Apparently, this doesn't work like the actual mount command. You can' t just prepend the server and have it magically be resolved like the linux mount utility would do.